types/occ/scheduler.go

```diff
@@ -18,3 +23,147 @@ func NewEstimateAbort(dependentTxIdx int) Abort {
 		Err:            ErrReadEstimate,
 	}
 }
+
+type TaskStatus string
+
+const (
+	TaskStatusPending   TaskStatus = "pending"
+	TaskStatusExecuted  TaskStatus = "executed"
+	TaskStatusAborted   TaskStatus = "aborted"
+	TaskStatusValidated TaskStatus = "validated"
+)
+
+type Task struct {
+	Status      TaskStatus
+	Index       int
+	Incarnation int
+	Request     types.RequestDeliverTx
+	Response    *types.ResponseDeliverTx
+}
```

Comment:
do we need to store any data here about "dependent TX" for the ESTIMATE abort case?

Reply:
yes once we do the estimate/validation pieces there are a few things we'll need to track